src/pages/search/components/advanced_search/ingredientsReducer.ts

```diff
@@ -0,0 +1,33 @@
+
+type IngredientsReducerAction =
+    | {
+          type: "add" | "remove";
+          value: string;
+      }
+    | {
+          type: "reset";
+      };
+
+const ingredientsReducer = (
+    ingredients: Set<string>,
```

Comment:
Chỗ này sử dụng `Set` để lưu nguyên liệu thì cũng đúng bài toán nhưng mà việc tạo lại `Set` để thể hiện sử thay đổi thì không chắc lắm về performance ...

Reply:
Vài chục item chắc ko có vấn đề gì đâu